src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -695,6 +682,12 @@ private void InitNativeProcess()
                         }
 #endif
                     }
+
+                    if (UpstreamIsNativeCommand)
+                    {
+                        _processInitialized ??= new SemaphoreSlim(0, 1);
+                        _processInitialized.Release();
+                    }
```

Comment:
btw I removed the lock because we're already in a lock for that object and I just didn't notice (line 596)